### DISCUSSION TITLE
Unexpected behaviour when use router push with URL object

### DISCUSSION THREAD

**Discussion by souzantero** [vercel/next.js#17625] [Help] [answered]
According to the Next.js [documentation](https://nextjs.org/docs/api-reference/next/router), I can use the router push function with a URL object to route dynamic paths, like this example :

```
import { useRouter } from 'next/router'

export default function ReadMore({ post }) {
  const router = useRouter()

  return (
    <span
      onClick={() => {
        router.push({
          pathname: '/post/[pid]',
          query: { pid: post.id },
        })
      }}
    >
      Click here to read more
    </span>
  )
}
```

But, when I try use this feature, the Next.js router sends the following `/post/[pid]?pid=1` to the browser (we take for example that the post id is 1), that is, the router does not replace the `[pid]` with the value of the query object.

Can someone help me?

**Comment by lfades** [comment, marked as answer]
@souzantero Our docs are a little ahead of the current Next.js version, your code should work on `next@canary`. We'll release a new version of Next.js soon that matches properly the docs.

**Reply by souzantero** [reply]
Thank you very much for the answer!

**Reply by flackeryy** [reply]
@lfades it doesn't work for me with `"next": "^10.0.4-canary.5",` and `"next": "^10.0.3"`. Am I doing something wrong?
